### PR TITLE
Login page

### PR DESCRIPTION
**Pull request by number10mechina**

### CODE REVIEW

File: Nahagos/app/src/main/java/com/nahagos/nahagos/MainActivity.java [hunk 1]
```diff
@@ -1,20 +1,103 @@
...
+        Button button = findViewById(R.id.loginButton);
+        button.setOnClickListener(v -> {
+
+            String username  = usernameObj.getText().toString().trim();
+            String password  = passwordObj.getText().toString().trim();
+            String driverId  = driverIdObj.getText().toString().trim();
+
+            if (!username.isEmpty() && !password.isEmpty()) {
+                Log.d("MainActivity", "not null");
+                Log.d("MainActivity", Boolean.toString(isCheckedGlobal));
```
Comment (goatOfLife) [inline Nahagos/app/src/main/java/com/nahagos/nahagos/MainActivity.java:77]: before merge, should remove logging info

File: Nahagos/app/src/main/java/com/nahagos/nahagos/MainActivity.java [hunk 2]
```diff
@@ -1,20 +1,103 @@
...
 
 import android.os.Bundle;
+import android.os.StrictMode;
+import android.util.Log;
+import androidx.appcompat.app.AppCompatActivity;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
```
Comment (goatOfLife) [inline Nahagos/app/src/main/java/com/nahagos/nahagos/MainActivity.java:12]: i believe some of these imports are not used here, and were here before you built Networks.java and ServerAPI. if true, please delete.

File: Nahagos/app/src/main/java/com/nahagos/nahagos/Networks.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    }
+
+    // Method for HTTP GET request
+    public static void httpGetReq(String urlString) {
+        HttpURLConnection connection = null;
+
+        try {
+            connection = setupConnection(urlString, "GET");
+            int responseCode = connection.getResponseCode();
+            Log.d(TAG, "Response Code: " + responseCode);
```
Comment (goatOfLife) [inline Nahagos/app/src/main/java/com/nahagos/nahagos/Networks.java:52]: again, although great for testing, logging should be removed before merge.

File: Nahagos/app/src/main/java/com/nahagos/nahagos/ServerAPI.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    public ServerAPI(Context context) {
+        ROOT_URL = "http://" + context.getString(R.string.server_ip) + ":8000";
+    }
+
+    public void passengerLogin(String username, String password)
+    {
+        String jsonBody = "{\"username\": \"" + username + "\", \"password\":\"" + password + "\"}";
+        Networks.httpPostReq(ROOT_URL + PASSENGER_LOGIN_URL, jsonBody);
+    }
+    public void driverLogin(String username, String password, String id)
```
Comment (goatOfLife) [inline Nahagos/app/src/main/java/com/nahagos/nahagos/ServerAPI.java:20]: Might want to validate we don't send false ids to the server, so if a driver tries to log in with id "@$%$#@" it shouldn't send anything to the server.

File: Nahagos/app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -44,4 +45,14 @@
...
     <string name="title_activity_main">MainActivity</string>
+    <string name="title_activity_register">register</string>
+    <string name="title_activity_login">LoginActivity</string>
+    <string name="prompt_email">Email</string>
+    <string name="prompt_password">Password</string>
+    <string name="action_sign_in">Sign in or register</string>
+    <string name="action_sign_in_short">Sign in</string>
+    <string name="welcome">"Welcome !"</string>
+    <string name="invalid_username">Not a valid username</string>
+    <string name="invalid_password">Password must be >5 characters</string>
```
Comment (goatOfLife) [inline Nahagos/app/src/main/res/values/strings.xml:56]: is it constant? might be a good idea to change it to a variable, the minimum length of password, for future changes. Not that critical.

**Review by goatOfLife — CHANGES_REQUESTED** [review]

File: Nahagos/app/src/main/java/com/nahagos/nahagos/Networks.java (hunk 3 above)
Comment (number10mechina) [reply Nahagos/app/src/main/java/com/nahagos/nahagos/Networks.java:52]: relevant for the next developers, theyll need to implement parser for http coming requests

File: Nahagos/app/src/main/java/com/nahagos/nahagos/ServerAPI.java (hunk 4 above)
Comment (number10mechina) [reply Nahagos/app/src/main/java/com/nahagos/nahagos/ServerAPI.java:20]: relevent for the next developers & the server response - both not implemented

File: Nahagos/app/src/main/java/com/nahagos/nahagos/MainActivity.java (hunk 2 above)
Comment (number10mechina) [reply Nahagos/app/src/main/java/com/nahagos/nahagos/MainActivity.java:12]: ok!

File: Nahagos/app/src/main/java/com/nahagos/nahagos/MainActivity.java (hunk 1 above)
Comment (number10mechina) [reply Nahagos/app/src/main/java/com/nahagos/nahagos/MainActivity.java:77]: ok!

File: Nahagos/app/src/main/res/values/strings.xml (hunk 5 above)
Comment (number10mechina) [reply Nahagos/app/src/main/res/values/strings.xml:56]: in the feature features

File: Nahagos/app/src/main/java/com/nahagos/nahagos/MainActivity.java (hunk 1 above)
Comment (goatOfLife) [reply Nahagos/app/src/main/java/com/nahagos/nahagos/MainActivity.java:59]: @number10mechina removed irrelevant logging, but some are still relevant for debugging later.

**Review by goatOfLife — APPROVED** [review]

**Pull request merged**